spec/std/http/server/server_spec.cr

```diff
@@ -151,6 +151,15 @@ module HTTP
     end
   end
 
+  describe HTTP::Server do
+    it "re-sets special port zero after listening" do
+      server = Server.new("0.0.0.0", 0) { |ctx| }
+      spawn { server.listen }
+      sleep 0.0001
```

Comment:
So, this `sleep 0.0001` will have to be present in user test suite too, if they want to launch server with random port in their test suite.

What do you think about this idea: split `#listen` in two parts: `#start` and `#listen`:

```crystal
def start
  return if @started
  @server = TCPServer.new(@host, @port)
  @port = server.local_address.port.to_i
  @started = true
end

def listen
  start
  until @wants_close
     spawn handle_client(server.accept)
   end
end
```

This will preserve current behavior if user calls only `#listen`. On the other hand it allows user to do:

```crystal
  server = Server.new("0.0.0.0", 0) { |ctx| }
  server.start
  port = server.port
  spawn { server.listen }
  server.port.should_not eq(0)
```

WDYT?